### PR TITLE
need location bool and error message

### PR DESCRIPTION
**Pull request by colehurwitz**
All sorters that need locations has `requires_locations` set to `True`. If the recording does not have locations in these cases, the sorter will fail with an informative error message upon initialization.

### CODE REVIEW

**Comment by colehurwitz** [comment]
This is passing... not sure what the issue is

File: spikesorters/kilosort/kilosort.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ class KilosortSorter(BaseSorter):
     sorter_name: str = 'kilosort'
     kilosort_path: Union[str, None] = os.getenv('KILOSORT_PATH', None)
     installed = check_if_installed(kilosort_path)
+    requires_locations = False
```
Comment (samuelgarcia) [inline spikesorters/kilosort/kilosort.py]: kilosort can play with geometry ??????

**Comment by samuelgarcia** [comment]
Hi COle I think I don't anderstand this PR.
In my mind absolutly all sorter need location (channel geometry).
No ?

**Comment by colehurwitz** [comment]
Not the case Samuel. Many sorters are fine without geometry and will substitute dummy channel. Only HerdingSpike and ironclust (so far) seem to fail without locations

**Comment by colehurwitz** [comment]
Checkout ms4 for example and it is fine without geometry. I think it is up to the sorter whether or not it needs **explicit** channel locations

**Comment by colehurwitz** [comment]
I was under the impression kilosort does not need geometry to run. If it does, we should change that boolean

**Comment by samuelgarcia** [comment]
What ahappen in tridesclous is that it make a fake linear geometry, which is a bad partice.
Distance between channels is need at some point in trideclous.
For tetorde, it do not matter at all but for mea it matter a lot.
I would propose to put True for tridesclous.

And for klusta i though it need also.

**Comment by colehurwitz** [comment]
Ok, we were targeting sorters that cannot sort without channel locations (at all). Herding spikes and ironclust try to do localization so they definitely need it. 

Do you want me to change the book for trideclous and Klusta then? They seem to still work even without locations so maybe we can just give a warning instead?

**Comment by colehurwitz** [comment]
Which sorters would put this Boolean to True for?

**Pull request merged**